scalameta/trees/shared/src/main/scala/scala/meta/internal/trees/ParentChecks.scala

```diff
@@ -105,6 +105,15 @@ object ParentChecks {
     tree.tpe.is[Type.Singleton] ==> (parent.is[Ctor.Secondary] && destination == "init")
   }
 
+  def EnumCase(tree: Tree, parent: Tree, destination: String): Boolean = {
+    parent.is[Template]
```

Comment:
Add check if the grandparent is non-empty, then it must be an enum

Reply:
Added.